web/src/main/java/org/springframework/security/web/util/matcher/RegexRequestMatcher.java

```diff
@@ -53,6 +54,26 @@ public final class RegexRequestMatcher implements RequestMatcher {
 
 	private final HttpMethod httpMethod;
 
+	/**
+	 * Creates a case-sensitive {@code Pattern} instance to match against the request.
+	 * @param pattern the regular expression to compile into a pattern.
+	 * @since 5.8
+	 */
+	public static RegexRequestMatcher forPattern(String pattern) {
+		Assert.hasText(pattern, "pattern cannot be empty");
+		return new RegexRequestMatcher(pattern, null);
+	}
+
+	/**
+	 * Creates an instance that matches to all requests with the same {@link HttpMethod}.
+	 * @param method the HTTP method to match. Must not be null.
+	 * @since 5.8
+	 */
+	public static RegexRequestMatcher forMethod(HttpMethod method) {
```

Comment:
Right now there is a problem if someone needs to do a static import for both AntPathRequestMatcher.forMethod and RegexRequestMatcher.forMethod the usage would need to be fully qualified. I'd say we should use antMatcher and regexMatcher for the method names.